src/shared/numa-util.c

```diff
@@ -124,6 +126,58 @@ int numa_to_cpu_set(const NUMAPolicy *policy, CPUSet *ret) {
         return 0;
 }
 
+static int numa_max_node(int *ret) {
+        _cleanup_closedir_ DIR *d = NULL;
+        struct dirent *de;
+        int r, max_node = 0;
+
+        assert(ret);
+
+        d = opendir("/sys/devices/system/node");
+        if (!d)
+                return -errno;
+
+        FOREACH_DIRENT(de, d, break) {
+                int node;
+                const char *n;
+
+                (void) dirent_ensure_type(d, de);
+
+                if (de->d_type != DT_DIR)
+                        continue;
+
+                n = startswith(de->d_name, "node");
+                if (!n)
+                        continue;
+
+                r = safe_atoi(n, &node);
```

Comment:
Shouldn't this be atou?

Reply:
it doesn't ultimately matter, since max_node is initialized to 0, and anything < 0 is not going to be larger than 0...

Reply:
Meh, maximum number of NUMA nodes is well below limit of `int`.